### PR TITLE
Update release publishing guide and move it to an issue template (Lombiq Technologies: OCORE-163)

### PR DESCRIPTION
**Pull request by Piedone**
Related to #15824. I updated the release publishing guide with what we actually do, and optimized it a bit. I also moved it to an issue template, since having to always copy it to an issue is unnecessary.

**Reviewers:** if you want to see what I changed in the guide before moving it to the issue template, see the diff [here](https://github.com/OrchardCMS/OrchardCore/pull/15830/files/f030fa490cacf20382c12ea276d1f1abf33bc15b).

## Summary by CodeRabbit

- **Documentation**
	- Updated the GitHub issue template for publishing Orchard Core releases.
	- Streamlined the release process documentation to guide users through a GitHub issue checklist.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+- [ ] Assign the milestone of the release's version to this issue.
+- [ ] Close remaining issues for the version (including merging corresponding pull requests if suitable) or assign them to the next one.
+- [ ] Assign all issues that were closed for an upcoming version (including a wildcard version like "1.0.x") to this version (milestone).
+
+### Prepare the code and documentation
+
+Update the source so everything looks like on the new version.
+
+- [ ] Create a `release/<version name>` (e.g. `release/1.8.0``) branch out of `main` or the previous release branch. This is needed so we can target pull requests on it.
```
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:24]: Keep " This is needed so we can target pull requests on it." though.

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    - [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/)
+- [ ] Update [Try Orchard Core](https://github.com/OrchardCMS/TryOrchardCore).
+
+### Publicize the release
+
+Let the whole world know about our shiny new release. Savor this part! These steps will make the release public so only do them once everything else is ready.
+
+- [ ] Tweet
+- [ ] Post to the [Orchard Core LinkedIn group](https://www.linkedin.com/groups/13605669/)
+- [ ] Post to the [Orchard Core Facebook page](https://www.facebook.com/OrchardCore/)
```
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:79]: There's no difference.

File: src/docs/topics/publishing-releases/README.md [hunk 3]
```diff
@@ -4,90 +4,8 @@ These notes are primarily for Orchard's core contributors to guide how to prepar
...
-- [ ] Post to the [Orchard Core Facebook page](https://www.facebook.com/OrchardCore/)
-
-### After the release is done
-
-- [ ] Create a new milestone with the next release number.
-- [ ] Create a new release notes documentation file for the next version in the OrchardCore.Docs project. (e.g, `/releases/1.8.0.md`).
-- [ ] Update the `OrchardCore.Commons.props` file with the next release number, and `<VersionSuffix>preview</VersionSuffix>` such that preview builds use the new one.
-
-```
+[Create a release issue](https://github.com/OrchardCMS/OrchardCore/issues/new/choose) and follow its checklist, ticking everything as you progress.
```
[the remark this answers is not in the document]
Comment (Piedone) [reply src/docs/topics/publishing-releases/README.md:11]: There's no difference.

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:24]: Update the suggestion, then.

File: .github/ISSUE_TEMPLATE/release.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+- [ ] Assign all issues that were closed for an upcoming version (including a wildcard version like "1.0.x") to this version (milestone).
+
+### Prepare the code and documentation
+
+Update the source so everything looks like on the new version.
+
+- [ ] Create a `release/<version name>` (e.g. `release/1.8.0`) branch out of `main` or the previous release branch. This is needed so we can target pull requests on it.
+- [ ] Create an issue branch out of this branch as usual.
+- [ ] Check the [release_ci](https://github.com/OrchardCMS/OrchardCore/blob/main/.github/workflows/release_ci.yml) workflow is using the expected .NET version to build the Docker images.
+- [ ] Update the `OrchardCore.Commons.props` file with `<VersionSuffix></VersionSuffix>` such that preview build numbers are not injected in packages. Verify the `VersionPrefix` tag matches the released version.
```
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:27]: That would be incorrect spelling with two spaces.

File: .github/ISSUE_TEMPLATE/release.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+- [ ] Update module versions in `src/OrchardCore/OrchardCore.Abstractions/Modules/Manifest/ManifestConstants.cs`.
+- [ ] Create a new milestone.
+- [ ] Create release notes in a specific documentation section. You can take the previous release notes as a template.
+    - Overview of the release's highlights and goals. What do you want people to remember this release for?
+    - Prerequisites. What framework version do you need, anything else to work with Orchard?
+    - Upgrade steps, any migration necessary from previous versions, breaking changes.
```
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:33]: These are sub-items and need to be indented.

[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:33]: Nope, the original code is good.

File: .github/ISSUE_TEMPLATE/release.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+- [ ] Assign all issues that were closed for an upcoming version (including a wildcard version like "1.0.x") to this version (milestone).
+
+### Prepare the code and documentation
+
+Update the source so everything looks like on the new version.
+
+- [ ] Create a `release/<version name>` (e.g. `release/1.8.0``) branch out of `main` or the previous release branch. This is needed so we can target pull requests on it.
+- [ ] Create an issue branch out of this branch as usual.
+- [ ] Check the [release_ci](https://github.com/OrchardCMS/OrchardCore/blob/main/.github/workflows/release_ci.yml) workflow is using the expected .NET version to build the Docker images.
+- [ ] Update the `OrchardCore.Commons.props` file with `<VersionSuffix></VersionSuffix>` such that preview build numbers are not injected in packages. Verify the `VersionPrefix` tag matches the released version.
```
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:27]: That would be an incorrect duplicated space.

File: .github/ISSUE_TEMPLATE/release.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+- [ ] Create a `release/<version name>` (e.g. `release/1.8.0`) branch out of `main` or the previous release branch. This is needed so we can target pull requests on it.
+- [ ] Create an issue branch out of this branch as usual.
+- [ ] Check the [release_ci](https://github.com/OrchardCMS/OrchardCore/blob/main/.github/workflows/release_ci.yml) workflow is using the expected .NET version to build the Docker images.
+- [ ] Update the `OrchardCore.Commons.props` file with `<VersionSuffix></VersionSuffix>` such that preview build numbers are not injected in packages. Verify the `VersionPrefix` tag matches the released version.
+- [ ] Update module versions in `src/OrchardCore/OrchardCore.Abstractions/Modules/Manifest/ManifestConstants.cs`.
+- [ ] Create a new milestone.
+- [ ] Create release notes in a specific documentation section. You can take the previous release notes as a template.
+    - Overview of the release's highlights and goals. What do you want people to remember this release for?
+    - Prerequisites. What framework version do you need, anything else to work with Orchard?
+    - Upgrade steps, any migration necessary from previous versions, breaking changes.
```
Comment (MikeAlhayek) [inline .github/ISSUE_TEMPLATE/release.md:33]:
```suggestion
    - Upgrade steps, any migration necessary from previous versions, and any breaking changes.
```

**Review by MikeAlhayek — APPROVED** [review]
LGTM

**Comment by Piedone** [comment]
Modified it a bit from https://github.com/OrchardCMS/OrchardCore/pull/15938.

File: .github/ISSUE_TEMPLATE/release.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
+---
+name: Publish a release
+about: Publish a new Orchard Core release
+title: 'Release v'
+labels: release
+assignees: ''
+
+---
+<!-- Be sure to also read https://docs.orchardcore.net/en/latest/topics/publishing-releases/. While the checklist is in a recommended order not every step depends strictly on the previous ones.  
+`<version name>` should be replaced with the current version, e.g. "1.8.0.". -->
```
[the remark this answers is not in the document]
Comment (Piedone) [reply .github/ISSUE_TEMPLATE/release.md:10]: This is a comment, so this shouldn't be changed.

**Comment by Piedone** [comment]
@coderabbitai pause

**Comment by Piedone** [comment]
@coderabbitai stop

**Pull request merged**